src/audio/copier/copier.c

```diff
@@ -1491,6 +1491,13 @@ static int copier_params(struct comp_dev *dev, struct sof_ipc_stream_params *par
 			if (dev->ipc_config.type == SOF_COMP_HOST && !cd->ipc_gtw) {
```

Comment:
sorry, could you explain? I thought this condition above meant "not an IPC gateway" and the commit description says "host gateway is used?"

Reply:
That "if" condition means "if Host gateway". Both Host and IPC gateway use SOF_COMP_HOST as a type. Maybe that is wrong and some specific type should be added for IPC gateway, but this is how it is now.

Reply:
clarified: in this terminology host and DAI components are called gateways too!

Reply:
please check ipc_gtw, if it is true, this part can't be entered with gateway case, it only cover host but not gateway case.